### ISSUE TITLE
vhs in backend partially broken in 3.0.0

### ISSUE THREAD

**Issue by Ceremony64** [FluidTYPO3/vhs#1066]
I use the DCE extension and defined custom backend views for the content elements.
For example, using the v:page.info viewhelper, I get the title of a page:
```
{namespace v=FluidTYPO3\Vhs\ViewHelpers}
{v:page.info(pageUid: '1', field: 'title')}
```

This no longer works in 3.0.0, despite working in 2.4.0 without any issues.
When saving a DCE element that has such a viewhelper, it will save it just fine (frontend shows the updated element), but it will result in an error for the backend view. PHP Error log shows this:
> PHP Fatal error:  __clone method called on non-object in /var/www/typo3conf/ext/vhs/Classes/Service/PageService.php on line 190

The line: https://github.com/FluidTYPO3/vhs/blob/development/Classes/Service/PageService.php#L190

I am running 7.6.9 LTS with DCE 1.3.6 with PHP5

**Comment by medienwerft** [comment]
A similar condition for me. TYPO3 v7.6.9, but without DCE. 

I use this view helper within the preview section of an fluidcontent element like:
`{v:page.info(pageUid: link, field: 'title') -> v:variable.set(name: 'headline')}
<h3>{headline}</h3>`

Same error.
![2016-07-14_093107](https://cloud.githubusercontent.com/assets/5450764/16831517/b7600548-49a5-11e6-98ff-a4fe477457b6.png)

**Comment by cedricziel** [comment]
We're currently discussing, whether that's a good plan at all.

The ViewHelper relies on the `TypoScriptFrontendController` which is not correctly initialized in the Backend. Maybe we should change the strategy here.

**Comment by NamelessCoder** [comment]
Some very relevant information about where TYPO3 may (should or must, imho) move to a completely different strategy regarding every Fluid ViewHelper that currently requires and simulates a frontend context:

https://forge.typo3.org/issues/73183

To summarise, the idea is to completely remove any and all frontend simulation from Fluid ViewHelpers and introduce a single new ViewHelper which allows wrapping template code that requires simulation.

The current process of simulating and tearing down simulated frontend contexts in the backend is insanely expensive (happens on *every* ViewHelper call) and may provide widely different results (for example, not being able to depend on TS configuration due to not having a page UID available) without the user having any clue as to which part of the hardcoded simulated frontend is incorrectly configured or missing.

If/when this happens that will also have an impact on VHS. At such time, when such a feature is included in the most recent TYPO3 version (that isn't LTS, or hopefully in the new LTS itself!) **we will also be removing each and every case of simulation or context checking from VHS ViewHelpers and require that context to be manually simulated via ViewHelper or in code that renders the template**.

What I mean to say with this is that we may not put a big effort into making the current ViewHelpers cross-context capable (because any such effort is expected to be wasted; scheduled for removal as soon as it gets added). It's not a nice message to give to users but *please do read and consider all the reasoning that is included in the TYPO3 Forge issue* - after which I hope you'll agree that our efforts are better spent in other directions ;)

**Comment by Ceremony64** [comment]
I understand your stance and that it is time consuming and inefficient to simulate the context in the backend. I like the idea of implementing it once to be used with vhs and other viewhelpers as a dependency, but unfortunately, we are not there yet:
This is a regression, so everyone who used certain viewhelpers in 2.4.0 and before will not be able to migrate to 3.0.0 without a regression in the rendered content.

**Comment by cedricziel** [comment]
Yes, @Ceremony64 .

We're aware that this issue might affect a small percentage of the user base (small as in: not all users do upgrade that fast and using `v:page.info` in the backend views seems rather unusual to me), and we already began solving it. (See the referenced PR above)

Cheers,
Cedric

**Comment by NamelessCoder** [comment]
Of course the message above doesn't mean we don't acknowledge there *is* a problem and won't fix it - I only wanted to indicate that it may not be top priority (at least for me, as I'd rather push to include the linked TYPO3 Forge change to make all of our lives easier). There will be a fix in 3.0.1 at least for this one!

**Comment by Ceremony64** [comment]
Thanks a lot guys! Looking forward to the fix!
